app/src/main/java/com/example/henzoshimada/feeltrip/MainScreen.java

```diff
@@ -150,11 +163,43 @@ public void onCheckedChanged(CompoundButton arg0, boolean isChecked) {
             }
         });
 
+        ToggleButton toggleMostRecent = (ToggleButton) findViewById(R.id.toggleMostRecent);
+        toggleMostRecent.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+
+            @Override
+            public void onCheckedChanged(CompoundButton arg0, boolean isChecked) {
+                //Toast.makeText(getApplicationContext(),"Toggled filter for friends!",Toast.LENGTH_SHORT).show();
+                FeelTripApplication.getFilterController().setMostrecentfilter(isChecked);
+                FeelTripApplication.loadFromElasticSearch();
+                FeelTripApplication.getMoodAdapter(getBaseContext()).notifyDataSetChanged();
```

Comment:
In that case, ElasticsearchController would make the most sense. We'll just refactor the segment then.